tests/http.rs

```diff
@@ -162,3 +175,293 @@ async fn test_http_server_reader_writer() {
     // start servers and don't stop them.
     terminate.send(()).unwrap();
 }
+
+#[test_case(
+    "csv",
+    true,
+    Some(true);
+    "CSV file schema supplied w/ headers")
+]
+#[test_case(
+    "csv",
+    true,
+    Some(false);
+    "CSV file schema supplied w/o headers")
+]
+#[test_case(
+    "csv",
+    false,
+    Some(true);
+    "CSV file schema inferred w/ headers")
+]
+#[test_case(
+    "csv",
+    false,
+    Some(false);
+    "CSV file schema inferred w/o headers")
+]
+#[test_case(
+    "parquet",
+    false,
+    None;
+    "Parquet file")
+]
+#[tokio::test]
+async fn test_upload_base(
+    file_format: &str,
+    include_schema: bool,
+    add_headers: Option<bool>,
+) {
+    let (addr, server, terminate, context) = make_read_only_http_server().await;
+
+    tokio::task::spawn(server);
+
+    let table_name = format!("{}_table", file_format);
+
+    // Prepare the schema + data (a single record batch) which we'll save to a temp file via
+    // a corresponding writer
+    let schema = Arc::new(Schema::new(vec![
+        Field::new("number", DataType::Int32, false),
+        Field::new("parity", DataType::Utf8, false),
+    ]));
+
+    // For CSV uploads we can supply the schema as another part of the multipart request, to
+    // remove the ambiguity resulting from automatic schema inference
```

Comment:
Out of interest, does it correctly infer `number: Int32` (or some numeric-looking value) when the schema is omitted?

Reply:
Oh wait, disregard this, it's actually checked in this very test 😅

Reply:
No it defaults to Int64 for integers, and similarly for other types: https://github.com/apache/arrow-rs/blob/master/arrow/src/csv/reader.rs#L87-L100